packages/apollos-church-api/src/data/rock-constants/index.js

```diff
@@ -0,0 +1,52 @@
+import { RockConstants as apollosRockConstants } from '@apollosproject/data-connector-rock';
+import ApollosConfig from '@apollosproject/config';
+
+const { ROCK_MAPPINGS } = ApollosConfig;
+
+const mapNewSpringNameToRockName = (name) => {
+  if (ROCK_MAPPINGS.PRAYER_REQUEST_TYPE.includes(name)) {
+    return 'PrayerRequest';
+  }
+  throw new Error(`${name} is not the correct Prayer Request Rock type!`);
+};
+
+class RockConstants extends apollosRockConstants.dataSource {
+  async prayerRequestInteractionComponent({
+    prayerRequestId,
+    prayerName = null,
+  }) {
+    const channel = await this.prayerRequestInteractionChannel();
+    return this.createOrFindInteractionComponent({
+      componentName: `${
+        ROCK_MAPPINGS.INTERACTIONS.COMPONENT_NAME_PRAYER_REQUEST
+      } - ${prayerName || prayerRequestId}`,
+      channelId: channel.id,
+      entityId: parseInt(prayerRequestId, 10),
+    });
+  }
+
+  async prayerRequestInteractionChannel() {
+    const { id } = await this.prayerModelType('PrayerRequest');
+    return this.createOrFindInteractionChannel({
+      channelName: ROCK_MAPPINGS.INTERACTIONS.CHANNEL_NAME,
```

Comment:
Why wouldn't this be `COMPONENT_NAME_PRAYER_REQUEST` as well?

Reply:
This indicates where the interaction is coming from, `NewSpring Apollos App`

Reply:
Oh. I read `COMPONENT_NAME` for some reason.